test/jdk/java/nio/channels/FileChannel/LargeMapTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.incubator.foreign.MemorySegment;
+import jdk.incubator.foreign.ResourceScope;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+
+/*
+ * @test
+ * @bug 8286637
+ * @requires os.family == "windows"
+ * @summary Ensure that memory mapping beyond 32-bit range does not cause an
+ *          EXCEPTION_ACCESS_VIOLATION.
+ */
+public class LargeMapTest {
+    private static final long LENGTH = 8000000000L;
+    private static final long OFFSET = 3704800000L;
+    private static final int  BUFSIZ = 100000;
+
+    public static void main(String[] args) throws IOException {
+        System.out.println(System.getProperty("sun.arch.data.model"));
+        System.out.println(System.getProperty("os.arch"));
+        System.out.println(System.getProperty("java.version"));
+
+        Path p = Path.of("test.dat");
+        RandomAccessFile raf = new RandomAccessFile("test.dat", "rw");
+        raf.setLength(LENGTH); //~8gb
+        raf.close();
+        p.toFile().deleteOnExit();
+
+
+        long offset = OFFSET;
+        ByteBuffer bb = ByteBuffer.allocateDirect(BUFSIZ);
+
+        MemorySegment mbb = MemorySegment.ofByteBuffer(bb);
+        MemorySegment mappedMemorySegment = MemorySegment.mapFile(p, 0,
+            p.toFile().length(), FileChannel.MapMode.READ_WRITE,
+            ResourceScope.newSharedScope());
```

Comment:
Note that I've integrated JEP 424. As a result, you will need to update this test in the following ways:

* replace `jdk.incubator.foreign` with `java.lang.foreign`
* add @enablePreview to the test
* replace `ResourceScope` with `MemorySession`
* the `map` method for segments has moved into `FileChannel`, for constistency

In addition, I think it would be better to use `MemorySession.openImplicit()` as session/scope parameter, since we don't want to call close explicitly.

Reply:
Would it be worth running the test on all platforms, rather than on windows only? And should we test other size combinations and offsets? For instance, Windows has some tricky code to split size and offset of the mapping - the test in this PR checks that the size is split correctly, but it doesn't check the mapping offset (which the test sets to 0).

Would be interesting perhaps to add a case which creates the mapping of a 8G file at offset 3G and then starts iterating.

Reply:
> Note that I've integrated JEP 424. As a result, you will need to update this test in the following ways:
> 
>     * replace `jdk.incubator.foreign` with `java.lang.foreign`
> 
>     * add @enablePreview to the test
> 
>     * replace `ResourceScope` with `MemorySession`
> 
>     * the `map` method for segments has moved into `FileChannel`, for constistency
> 
> 
> In addition, I think it would be better to use `MemorySession.openImplicit()` as session/scope parameter, since we don't want to call close explicitly.

Fixed by commit b5c3b0d9c25d3fcdee88f3e4f2b36c26e0de3d17.